### PR TITLE
[fix][broker] Fix delete system topic clean topic policy

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
If users set topic policy for system topic, then delete this system topic, the topic policy should be deleted.

### Modification
Only change_events topic do not need to clear topic policies.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/liangyepianzhou/pulsar/pull/16

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 1]
```diff
@@ -1221,8 +1221,7 @@ private CompletableFuture<Void> delete(boolean failIfHasSubscriptions,
 
                         deleteTopicAuthenticationFuture.thenCompose(ignore -> deleteSchema())
                                 .thenCompose(ignore -> {
-                                    if (!this.getBrokerService().getPulsar().getBrokerService()
-                                            .isSystemTopic(TopicName.get(topic))) {
+                                    if (!SystemTopicNames.isTopicPoliciesSystemTopic(topic)) {
```
Comment (congbobo184) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:1224]: topic policy system topic also can delete it topic policy, why we don't clean up the topic policy of the topic policy

Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:1224]: The topic policy topic will send a message to itself and then report `topic is fenced`.
And delete topic policy topic does not need to delete its policy.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java [hunk 2]
```diff
@@ -1616,4 +1616,33 @@ public void testGetTxnState() throws Exception {
         Transaction abortingTxn = transaction;
         Awaitility.await().until(() -> abortingTxn.getState() == Transaction.State.ABORTING);
     }
+    @Test
+    public void testDeleteTopicPolicyWhenDeleteSystemTopic() throws Exception {
```
Comment (congbobo184) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java:1620]: please move this test to [NamespacesTest.java](https://github.com/apache/pulsar/pull/18823/files#diff-24788d54292c43e74fe60920b6edd12c0526b6cf5808aec9ac5a134c554db31a)

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 3]
```diff
@@ -224,18 +225,28 @@ protected CompletableFuture<Void> internalDeleteNamespaceAsync(boolean force) {
...
                     List<String> allSystemTopics = new ArrayList<>();
                     List<String> allPartitionedSystemTopics = new ArrayList<>();
+                    List<String> topicPolicy = new ArrayList<>();
+                    List<String> partitionedTopicPolicy = new ArrayList<>();
                     for (String topic : allTopics) {
                         if (!pulsar().getBrokerService().isSystemTopic(TopicName.get(topic))) {
                             hasNonSystemTopic = true;
-                            allUserCreatedTopics.add(topic);
+                            if (SystemTopicNames.isTopicPoliciesSystemTopic(topic)) {
+                                topicPolicy.add(topic);
```
[the remark this answers is not in the document]
Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:236]: Right.

**Review by congbobo184 — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 4]
```diff
@@ -224,20 +225,30 @@ protected CompletableFuture<Void> internalDeleteNamespaceAsync(boolean force) {
                     boolean hasNonSystemTopic = false;
                     List<String> allSystemTopics = new ArrayList<>();
                     List<String> allPartitionedSystemTopics = new ArrayList<>();
+                    List<String> topicPolicy = new ArrayList<>();
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:228]: why not named `topicPolicySystemTopic` or `changeEventTopic`?

**Review by poorbarcode — APPROVED** [review]

**Comment by poorbarcode** [comment]
Link to

- https://github.com/apache/pulsar/pull/18307

**Review by codelipenghui — APPROVED** [review]

**Comment by Technoboy-** [comment]
Cherry-picked by #19835

**Pull request merged**